components/event-processor-manager/org.wso2.carbon.event.processor.manager.commons/src/main/java/org/wso2/carbon/event/processor/manager/commons/transport/client/TCPEventPublisher.java

```diff
@@ -260,23 +274,28 @@ public void onEvent(ByteArrayHolder byteArrayHolder, long sequence, boolean endO
 
     /**
      * Gracefully shutdown the TCPEventPublisher.
-     * When this method is used already consumer threads of distruptor will try to publishToDisruptor the queued messages in the RingBuffer.
+     * When this method is used already consumer threads of disruptor will try to publishToDisruptor the queued messages in the RingBuffer.
      */
-    public void shutdown() {
+    public void shutdown(boolean flushDataInBuffer) {
```

Comment:
Can we convert this to shutdown() and shutdownNow() which will not flush data. sending a boolean in the shutdown is not intuitive